frontend/ui/wikipedia.lua

```diff
@@ -380,7 +380,7 @@ function Wikipedia:createEpub(epub_path, page, lang, with_images, progress_callb
             table.insert(images, cur_image)
             seen_images[src] = cur_image
             -- Use first image of reasonable size (not an icon) and portrait-like as cover-image
-            if cover_imgid == "" and tonumber(width) > 50 and tonumber(height) > 50 and tonumber(height) > tonumber(width) then
+            if cover_imgid == "" and width and tonumber(width) > 50 and height and tonumber(height) > 50 and tonumber(height) > tonumber(width) then
```

Comment:
That could still crash.

```
tonumber("some random stuff that doesn't turn into a number will also evaluate be nil")
```

So you should do something more like

local width = tonumber(img_tag:match([[width="([^"]*)"]]))

type(width) == "number" and width > 50

Reply:
With the tonumber up there you don't need all of those tonumbers in here. `if width and width > 50` is safe now.

Reply:
Indeed...